jscomp/super_errors/super_typecore.ml

```diff
@@ -276,6 +292,20 @@ let report_error env ppf = function
         name (*kind*) Printtyp.path p;
        end;
       spellcheck ppf name valid_names;
+  | Name_type_mismatch (
+      "record",
+      Ldot (Ldot ((Lident "Js"), "Fn"), _arityFieldName),
+      (
+        _,
+        (Pdot (Pdot ((Pident {name = "Js"}), "Fn", _), arityA, _))
+      ),
+      [(
+        _,
+        (Pdot (Pdot ((Pident {name = "Js"}), "Fn", _), arityB, _))
+      )]
+    ) ->
```

Comment:
This pattern is a bit unclear to me. Here's the original: https://github.com/rescript-lang/ocaml/blob/bc2625fe99e2e01700d1ec0fea8dcc0672c94564/typing/typecore.ml#L5011

`tp` is a tuple of `Path.t`, and in my test here, both items are the same. `tpl` is a list of such tuple, and in my test here, a list with a single item of a tuple.

From what I understand, the second item of the tuple is the expanded path... right? So I should be matching on that and ignoring the first item?

Reply:
For the record:
```
fun[@bs] a0 a1 -> a0 + a1  
```
is expanded into 
```
{Js.Fn.I2 = fun a0 a1 -> a0 + a1}
```
it does not come with an annotation like below:
```
({Js.Fn.I2 = fun a0 a1 -> a0 + a1} : _ Js.Fn.arity2)
```
With such annotation, the error message is better.
However, this is intentional, since adding such annotation will make this
```
data -> mapU (fun [@bs] x -> x.label) 
```
not work

Reply:
To answer your question, `tp` is its type and expanded type (e.g, type aliases)
The changes are mostly correct, I will merge it and do some tweak later